### DISCUSSION TITLE
Why would a publish() request return false?

### DISCUSSION THREAD

**Discussion by CommuniG8** [uNetworking/uWebSockets.js#937] [Q&A]
Sorry very new to this.

I'm using javascript via node.
uWebSockets.js is v20.30.0 and node is v18.16.1

subscribe() is working for me all the time but publish() always returns false.
publish() has two string arguments.

**Comment by CommuniG8** [comment]
After some more testing...

Ruling out incorrect parameters to ws.publish(), a return value of false seems to mean that nobody is subscribed the the topic. So a ws.send() could not be performed  from ws.publish().

The publisher has to be subscriber to some topic or any ws.publish() will fail, does not matter what the topic is.
If the subscriber used a '#' wildcard e.g. 'home/stuff/#' then a ws.publish() to 'home/stuff/data' fails.
if the subscriber used a non-wildcard topic e.g. 'home/stuff/data' then a ws.publish() to 'home/stuff/data' works!

So first I don't under the need for the published to subscribe to a random topic.
Also wildcards don't work.
Or have I totally misunderstood this?

**Comment by e3dio** [comment]
v20 simplified and improved performance of Publish, removed wildcard topics: 
https://github.com/uNetworking/uWebSockets/releases/tag/v20.0.0

ws.publish() skips send to self, app.publish() sends to all

**Reply by CommuniG8** [reply]
OK that explains wildcards.
But why is a subscribe() required before a publish() can work?
